File: order-service/src/main/java/com/paulopacifico/orderservice/messaging/api/OrderPlacedEvent.java
package com.paulopacifico.orderservice.messaging.api;

import java.math.BigDecimal;
import java.time.OffsetDateTime;
import java.util.UUID;

public record OrderPlacedEvent(
        UUID eventId,
        Long orderId,
        String orderNumber,
        String skuCode,
        BigDecimal price,
        Integer quantity,
        OffsetDateTime occurredAt
) {
}

File: inventory-service/src/main/kotlin/com/paulopacifico/inventoryservice/messaging/api/OrderPlacedEvent.kt
package com.paulopacifico.inventoryservice.messaging.api

import java.math.BigDecimal
import java.time.OffsetDateTime
import java.util.UUID

data class OrderPlacedEvent(
    val eventId: UUID,
    val orderId: Long,
    val orderNumber: String,
    val skuCode: String,
    val price: BigDecimal,
    val quantity: Int,
    val occurredAt: OffsetDateTime,
)
